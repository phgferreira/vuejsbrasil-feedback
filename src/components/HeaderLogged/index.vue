<template>
  <div class="flex items-center justify-between w-4/5 max-w-6xl py-10">
    <div class="w-28 lg:w-36">
      <img class="w-full" src="../../assets/img/logo_white.png" alt="logo">
    </div>
    <div class="flex">
      <ul class="flex list-none">
        <li
          class="px-6 py-2 mr-2 font-bold text-white rounded-full  cursor-pointer focus:outline-none"
          @click="() => router.push({ name: 'Credencials' })"
        >
          Credencials
        </li>
        <li
          class="px-6 py-2 mr-2 font-bold text-white rounded-full  cursor-pointer focus:outline-none"
          @click="() => router.push({ name: 'Feedbacks' })"
        >
          Feedbacks
        </li>
        <li
          class="px-6 py-2 font-bold bg-white rounded-full cursor-pointer text-brand-main focus:outline-none"
          @click="handleLogout"
        >
          {{ logoutLabel }}
        </li>
      </ul>
    </div>
  </div>
</template>

<script>
import { useRouter } from 'vue-router'
import useStore from '@/hooks/useStore'
import { computed } from 'vue'
import { cleanCurrentUser } from '@/store/user'

export default {
  setup () {
    const router = useRouter()
    const store = useStore('User')

    const logoutLabel = computed(() => {
      return (!store.currentUser.name) ? '...' : `${store.currentUser.name} (sair)`
    })

    function handleLogout () {
      window.localStorage.removeItem('token')
      cleanCurrentUser()
      router.push({ name: 'Home' })
    }

    return {
      router,
      logoutLabel,
      handleLogout
    }
  }
}
</script>
