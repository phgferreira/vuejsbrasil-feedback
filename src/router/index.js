import { createRouter, createWebHistory } from 'vue-router'

export const routes = [
  {
    path: '/',
    name: 'Home',
    component: () => import('../views/home/index.vue')
  },
  {
    path: '/feedbacks',
    name: 'Feedbacks',
    component: () => import('../views/feedbacks/index.vue'),
    meta: {
      hasAuth: true
    }
  },
  {
    path: '/credencials',
    name: 'Credencials',
    component: () => import('../views/credencials/index.vue'),
    meta: {
      hasAuth: true
    }
  },
  {
    path: '/:pathMatch(.*)*',
    redirect: { name: 'Home' }
  }
]

const router = createRouter({
  history: createWebHistory('/'),
  routes
})

export default router
