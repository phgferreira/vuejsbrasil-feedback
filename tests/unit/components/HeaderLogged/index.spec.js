import { shallowMount } from '@vue/test-utils'
import HeaderLogged from '@/components/HeaderLogged/index.vue'
import { createRouter, createWebHistory } from 'vue-router'
import { routes } from '@/router'

const router = createRouter({
  history: createWebHistory('/'),
  routes
})

const mockStore = { currentUser: {} }
jest.mock('@/hooks/useStore', () => {
  return () => mockStore
})

describe('<HeaderLogged />', () => {
  it('deve renderizar o cabeçalho quando logado corretamente', async () => {
    await router.push('/')
    await router.isReady()
    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugings: [router]
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('deve renderizar três pontos quando usuário não estiver logado', async () => {
    await router.push('/')
    await router.isReady()
    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugings: [router]
      }
    })
    const buttonLogout = wrapper.find('[data-test="logout-button"]')
    expect(buttonLogout.text()).toEqual('...')
  })

  it('deve renderizar usuário quando estiver logado', async () => {
    await router.push('/')
    await router.isReady()
    mockStore.currentUser.name = 'Fulano'
    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugings: [router]
      }
    })
    const buttonLogout = wrapper.find('[data-test="logout-button"]')
    expect(buttonLogout.text()).toEqual('Fulano (sair)')
  })
})
