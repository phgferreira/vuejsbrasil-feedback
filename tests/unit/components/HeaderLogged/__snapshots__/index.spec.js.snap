// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderLogged /> deve renderizar o cabe√ßalho quando logado corretamente 1`] = `
<div class="flex items-center justify-between w-4/5 max-w-6xl py-10">
  <div class="w-28 lg:w-36"><img class="w-full" src="" alt="logo"></div>
  <div class="flex">
    <ul class="flex list-none">
      <li class="px-6 py-2 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none"> Credencials </li>
      <li class="px-6 py-2 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none"> Feedbacks </li>
      <li class="px-6 py-2 font-bold bg-white rounded-full cursor-pointer text-brand-main focus:outline-none" data-test="logout-button">...</li>
    </ul>
  </div>
</div>
`;
